### DISCUSSION TITLE
Unusual amount of inbound packets

### DISCUSSION THREAD

**Discussion by fuyutsuki** [WaterdogPE/WaterdogPE#110] [Q&A]
Hi, I have started a server using waterdogPE, but on the server running waterdogPE, inbound packets are 10 times larger than the outbound packets, causing the server to lag.
Do you have any ideas?

downstream server is pmmp and here's [my config](https://discord.com/channels/767330242078834712/767331674245431306/867781693955768390)

**Comment by TobiasGrether** [comment]
Hey there!
First of all, where are you getting these statistics from?
Second of all, does inbound refer to packets from the client to the proxy, from the proxy to the downstream server, or vise versa?

**Reply by fuyutsuki** [reply]
Hi @TobiasG-DE!

The statistics are from the web console of VPS, which averages around 4Mbps when waterdogPE is not used. However, when waterdogPE is used, the inbound rate swells to about 40 Mbps.  
Inbound here is the sum of the two: client to proxy and downstream to proxy.

**Reply by TobiasGrether** [reply]
The proxy should not use any bandwidth when it's not in use. Can you please verify this issue without any plugins?

**Reply by fuyutsuki** [reply]
I'll check it out.

**Reply by belohnung** [reply]
Does it only happen with players online? Maybe the VPS panel also counts traffic on the virtual interfaces of the server, eg. loopback where your downstream servers probably reside, making the vps panel see almost every packet twice

**Reply by fuyutsuki** [reply]
> The proxy should not use any bandwidth when it's not in use. Can you please verify this issue without any plugins?

The same issue occurred without the plugin, causing lag

**Reply by fuyutsuki** [reply]
> Does it only happen with players online? Maybe the VPS panel also counts traffic on the virtual interfaces of the server, eg. loopback where your downstream servers probably reside, making the vps panel see almost every packet twice

Maybe so, but that doesn't explain the more than fourfold increase in traffic…

**Reply by Alemiz112** [reply]
Can you clarify what does your "inbound traffic" state for? As Tobias mentioned, inbound traffic is the traffic between players and the proxy. In that case increased traffic is not caused by the proxy itself. It is most likely the fact that all traffic has to go thought proxy and than it is split into several streams flowing to downstream.

**Reply by belohnung** [reply]
> 
> 
> > Does it only happen with players online? Maybe the VPS panel also counts traffic on the virtual interfaces of the server, eg. loopback where your downstream servers probably reside, making the vps panel see almost every packet twice
> 
> Maybe so, but that doesn't explain the more than fourfold increase in traffic…

if i understood the code correctly, there is a downstream connection for every player meaning every player produces roughly 2x the traffic it would normally produce, thats not real traffic though, if the downstream servers are hosted on the same machine

**Reply by belohnung** [reply]
also i dont think 40mbps would cause lag

**Reply by fuyutsuki** [reply]
> also i dont think 40mbps would cause lag

Yes, that is the actual problem and I'd like to find out the cause of the lag.
So I asked this question here because this problem only occurs when using waterdogPE.

**Reply by Alemiz112** [reply]
You still haven't provided any details about what downstream software you use, how exactly does your network architecture looks like ... It is hard to say where the issue comes from without much details and guessing isn't the solution.

**Reply by fuyutsuki** [reply]
I wasn't sure what kind of information you needed, so I'll look into those first and let you know.

**Reply by fuyutsuki** [reply]
Upstream:
- OS: Fedora 34(x64)
- Arch: amd64
- Spec: 4-Core, 4GB RAM, 256GB SSD
- Software: WaterdogPE 

Downstream:
- OS: Ubuntu 21.04(x64)
- Arch: amd64
- Spec: 4-Core, 4GB RAM, 1TB SSD
- Software: pmmp 3.22.0
